### ISSUE TITLE
[Usage] can not print weight in callback function

### ISSUE THREAD

**Issue by gaojianzhang** [haotian-liu/LLaVA#1751]
### Describe the issue

Issue:
I have written a callback funcion to supervise the grad. But all of the outputs are tensor:[] and grad: None. How can I print the true parameters.
Command:
```
def make_supervised_data_module(tokenizer: transformers.PreTrainedTokenizer,
                                data_args) -> Dict:
    """Make dataset and collator for supervised fine-tuning."""
    train_dataset = LazySupervisedDataset(tokenizer=tokenizer,
                                data_path=data_args.data_path,
                                data_args=data_args)
    data_collator = DataCollatorForSupervisedDataset(tokenizer=tokenizer)
    return dict(train_dataset=train_dataset,
                eval_dataset=None,
                data_collator=data_collator)

def print_gradients(model):
    """
    递归遍历模型中的所有参数并打印它们的梯度。
    """
    model = model.base_model.model
    for name, param in model.named_parameters():
        print(f"{name}: Weight = {param.data}, Gradient = {param.grad}")

```

Log: 
```
model.layers.27.self_attn.v_proj.base_layer.weight: Weight = tensor([], device='cuda:0', dtype=torch.bfloat16), Gradient = None
model.layers.27.self_attn.v_proj.lora_A.default.weight: Weight = tensor([], device='cuda:0', dtype=torch.bfloat16), Gradient = None
model.layers.27.self_attn.v_proj.lora_B.default.weight: Weight = tensor([], device='cuda:0', dtype=torch.bfloat16), Gradient = None
```

**Comment by LiHaoHN** [comment]
Hello, have you found the cause of the problem? I have also encountered this problem.

**Comment by Pixel-anter** [comment]
Hello, have you found the cause of the problem? I have also encountered this problem.

**Comment by Pixel-anter** [comment]
You can get a global gradient print。
